simpleclient_caffeine/src/main/java/io/prometheus/client/cache/caffeine/CacheMetricsCollector.java

```diff
@@ -99,7 +117,10 @@ public void clear(){
     @Override
     public List<MetricFamilySamples> collect() {
         List<MetricFamilySamples> mfs = new ArrayList<MetricFamilySamples>();
-        List<String> labelNames = Arrays.asList("cache");
+        List<String> labelNames = new ArrayList<String>(customLabels.size() + 1) {{
+            add("cache");
+            addAll(customLabels);
+        }};
```

Comment:
Should be initialize this this List each time? Maybe simply initialize it once as instance field?